src/plugins/exec/sim-host-handlers.js

```diff
@@ -56,10 +58,13 @@ module.exports = {
 
                         dialog.hideDialog('exec-dialog');
 
-                        if (document.getElementById('exec-persist').checked) {
-                            savedSims.addSim({service: service, action: action, args: args, value: result, success: isSuccess});
+                        if (shouldPersist) {
+                            savedSims.addSim({ service: service, action: action, args: args, value: result, success: isSuccess });
                         }
 
+                        var hasValue = typeof result !== 'undefined' && result !== '';
```

Comment:
Yes, and that means that the user typed `null` in the popup dialog. In that case, I think that we should consider `hasValue` to be true, because the user entered a value, and we are interested in how users use the UI.

On the other hand, the current implementation prevents users from typing `undefined` in the popup, so the current check works in that case (if `typeof result === 'undefined'` then it truly means the user didn't type anything).